maintenance/wikia/deleteImageRevision.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+require_once( __DIR__ . '/../Maintenance.php' );
+
+class DeleteImageRevision extends Maintenance {
+
+	const PAGE_ID_OPTION = 'pageId';
+	const REVISION_ID_OPTION = 'revisionId';
+	const REASON_OPTION = 'reason';
+
+	public function __construct() {
+		parent::__construct();
+		$this->addOption( self::PAGE_ID_OPTION, "image page id", true );
+		$this->addOption( self::REVISION_ID_OPTION, "revision id", true );
+		$this->addOption( self::REASON_OPTION, "deletion reason", true );
+		$this->mDescription = "Remove given revision of given image file";
+	}
+
+
+	public function execute() {
+		$pageId = intval( $this->getOption( self::PAGE_ID_OPTION ) );
+		$revisionId = intval( $this->getOption( self::REVISION_ID_OPTION ) );
+		$reason = $this->getOption( self::REASON_OPTION );
+		$title = Title::newFromID( $pageId );
+
+		$db = wfGetDB( DB_SLAVE );
+		$timestamp = $db->selectField(
+			[ 'revision' ],
+			'rev_timestamp',
+			[
+				'rev_page' => $title->getArticleID(),
+				'rev_id' => $revisionId,
+			],
+			__METHOD__
+		);
+
+		if ( $this->isOldImageRevision( $db, $timestamp, $title->getDBkey() ) ) {
+			$this->removeOldRevision( $title, $timestamp, $revisionId, $pageId, $reason );
+		} else {
+			$this->removeLatestRevision( $title, $timestamp, $revisionId, $pageId, $reason );
+		}
+	}
+
+	private function removeLatestRevision( Title $title, string $revisionTimestamp, int $revisionId, int $pageId, string $reason ) {
+		if ($this->isOnlyRevision()) {
+			LocalFile::newFromTitle( $title, RepoGroup::singleton()->getLocalRepo() );
+			$oldfile = '';
+			if ( !FileDeleteForm::doDelete( $title, $file, $oldfile, $reason, false )->isOK() ) {
+				\Wikia\Logger\WikiaLogger::instance()->error(
+					"deleting file was not successful",
+					[
+						'page_id' => $pageId,
+						'revision_id' => $revisionId
+					]
+				);
+			}
+		} else {
+			// TODO: revert
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)